packages/SwingSet/src/kernel/gc-actions.js

```diff
@@ -1,17 +1,33 @@
-import { assert } from '@agoric/assert';
+import { Fail } from '@agoric/assert';
 import { insistKernelType } from './parseKernelSlots.js';
 import { insistVatID } from '../lib/id.js';
 
-const typePriority = ['dropExport', 'retireExport', 'retireImport'];
+const { ownKeys } = Reflect;
+
+/**
+ * A descending-priority mapping of GC action type to queue event type.
+ *
+ * @enum {string}
+ */
+const typePriority = {
```

Comment:
I'm not against this change from list to object/record, but.. is it a win? Seems like it's adding more code than it's removing, and the need for `setPrototypeOf(.. null)` is a smell. If it helps us detect typos in the `dropExport`/etc strings, then I can see it being a benefit, but I think we had that already (at the cost of an O(3) lookup).

Reply:
The smell I'm addressing here is ``` `${type}s` ```, which isn't a typo hazard as much as it is a grep dodge... better to explicitly show the map from one domain to another in my opinion. And the `setPrototypeOf` isn't necessary, just a way to be explicit (a reasonable alternative would omit it and replace later use of `in` with an own property check).

That said, I can revert back to the array if you feel better about that. Thoughts?

Reply:
Agreed that `${type}s` is a smell.

Using a record or Map as the primary list doesn't make it clear that the priority ordering of these three types is critical. Your implementation does the right thing, but (speaking from personal experience) most programmers/auditors don't automatically know that property enumeration happens in insertion order.

So how about:

```js
// priority of GC action types
const priorities = ['dropExport', 'retireExport', 'retireImport'];

// map GC action type to queue event type
const actionTypeToQueueEventType = new Map([
  [ 'dropExport', 'dropExports' ],
  [ 'retireExport', 'retireExports' ],
  [ 'retireImport', 'retireImports' ],
]);

...

assert(actionTypeToQueueEventType.has(type), type);

..

for (const type of priorities) {

..

const queueType = actionTypeToQueueEventType.get(type);
return harden({ type: queueType, ..
```